prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/repository/NodesClient.java

```diff
@@ -123,14 +123,9 @@ public void configure()
     @SuppressWarnings("PMD.UnusedPrivateMethod")
     private void setAuthorizationHeader(Exchange exchange)
     {
-        if (oAuth2ClientProperties.getProvider().containsKey("alfresco"))
-        {
-            exchange.getIn().setHeader(AUTHORIZATION, "Bearer " + accessTokenProvider.getAccessToken("alfresco"));
-        }
-        else
-        {
-            exchange.getIn().setHeader(AUTHORIZATION, getBasicAuthenticationHeader());
-        }
+        final String token = oAuth2ClientProperties.getProvider().containsKey("alfresco") ? "Bearer " + accessTokenProvider.getAccessToken("alfresco") : getBasicAuthenticationHeader();
```

Comment:
I really don't like this part because it's really low-level and tbh - I don't understand what's going on, I would have to really dive deep to understand it. Is there perhaps a way to make it simpler/more obvious?

Reply:
I've refactored that part - let me know if that helps.

Reply:
looks much better now! thanks